src/m365/spo/commands/contenttype/contenttype-sync.ts

```diff
@@ -0,0 +1,194 @@
+  }
+
+  public get description(): string {
+    return 'Adds a published content type from the content type hub to a site or syncs its latest changes';
+  }
+
+  constructor() {
+    super();
+
+    this.#initTelemetry();
+    this.#initOptions();
+    this.#initValidators();
+    this.#initTypes();
+    this.#initOptionSets();
+  }
+
+  #initTelemetry(): void {
+    this.telemetry.push((args: CommandArgs) => {
+      Object.assign(this.telemetryProperties, {
+        id: typeof args.options.id !== 'undefined',
+        name: typeof args.options.name !== 'undefined',
+        listId: typeof args.options.listId !== 'undefined',
+        listTitle: typeof args.options.listTitle !== 'undefined',
+        listUrl: typeof args.options.listUrl !== 'undefined'
+      });
+    });
+  }
+
+  #initOptions(): void {
+    this.options.unshift(
+      {
+        option: '-u, --webUrl <webUrl>'
+      },
+      {
+        option: '-i, --id [id]'
+      },
+      {
+        option: '-n, --name [name]'
+      },
+      {
+        option: '--listTitle [listTitle]'
+      },
+      {
+        option: '--listId [listId]'
+      },
+      {
+        option: '--listUrl [listUrl]'
+      }
+    );
+  }
+
+  #initValidators(): void {
+    this.validators.push(
+      async (args: CommandArgs) => {
+        const isValidSharePointUrl: boolean | string = validation.isValidSharePointUrl(args.options.webUrl);
+        if (isValidSharePointUrl !== true) {
+          return isValidSharePointUrl;
+        }
+
+        if (args.options.listId && !validation.isValidGuid(args.options.listId)) {
+          return `${args.options.listId} is not a valid GUID`;
+        }
+
+        return true;
+      }
+    );
+  }
+
+  #initTypes(): void {
+    this.types.string.push('webUrl', 'id', 'name', 'listTitle', 'listId', 'listUrl');
+  }
+
+  #initOptionSets(): void {
+    this.optionSets.push(
+      {
+        options: ['id', 'name']
+      },
+      {
+        options: ['listId', 'listTitle', 'listUrl'],
+        runsWhen: (args) => args.options.listId || args.options.listTitle || args.options.listUrl
+      }
+    );
+  }
+
+  public async commandAction(logger: Logger, args: CommandArgs): Promise<void> {
+    if (this.verbose) {
+      await logger.logToStderr(`Adding the content type ${args.options.name || args.options.id} to site ${args.options.webUrl} or syncing its latest changes...`);
+    }
+    try {
+      const url: URL = new URL(args.options.webUrl);
+
+      const baseUrl = 'https://graph.microsoft.com/v1.0/sites/';
+      const siteUrl = url.pathname === '/' ? url.host : await spo.getSiteId(args.options.webUrl, logger, this.verbose);
+
+      let listUrl = '';
+
+      if (args.options.listId || args.options.listTitle || args.options.listUrl) {
+        listUrl = `/lists/${args.options.listId || args.options.listTitle || await this.getListIdByUrl(args.options.webUrl, args.options.listUrl!, logger)}`;
+      }
+
+      if (this.verbose) {
```

Comment:
This verbose log line is either superfluous or should be moved to the getContentTypeId function.

Let's return `options.id` from there as well, like we often do.